docker/README.md

```diff
@@ -0,0 +1,128 @@
+# Dockerizing
+
+This video is a good resource for tips on how to optimize docker setup
+
+https://youtu.be/kG2vxYn547E?t=255 
+
+TLDR;
+
+[See slides here](https://speakerd.s3.amazonaws.com/presentations/55d8979b402d43f4a5a99cce56273579/containers.pdf)
+
+## Tip #1: Read and understand your base image
+
+There is an official [ruby image](https://hub.docker.com/_/ruby) on Docker Hub, but it comes with batteries included. 
+
+[Alpine](https://hub.docker.com/_/alpine) is a very slim linux that can be extended with ruby.
+
+## Tip #2: Combine install commands with cleanup
+
+👍👍👍
+```Docker
+RUN apt-get update -y \
+ && apt-get install -y -q mypackage \
+ && apt-get clean \
+ && rm -f /var/lib/apt/lists/*_* 
+```
+
+👎🤨😔
+```Docker
+RUN apt-get update -y
+RUN apt-get install -y -q mypackage
+RUN apt-get clean \
+ && rm -f /var/lib/apt/lists/*_* 
+```
+
+## Tip #3: Use a separate build stage
+
+Multi-stage docker files
+
+```Docker
+FROM my-base AS builder
+COPY . /app
+RUN apt-get install build-essential \
+ && bundle install --deployment
+
+FROM my-base
+COPY --from=builder /app /app
+```
+
+<img src="multi-stage-docker-files.png" alt="Multi-stage docker files" height="100">
+
+## Tip #4: Set the system locale
+
+```Docker
+RUN echo "en_US.UTF-8 UTF-8" > /etc/locale.gen \
+ && locale-gen en_US.UTF-8
+ENV LANG en_US.UTF-8
+ENV LANGUAGE en_US:en 
+```
+
+## Tip #5: Create an unprivileged user
+
+After building your app...
+
+```Docker
+RUN adduser -s /bin/sh -u 1001 -G root \
```

Comment:
why do we need this?

Reply:
It is best practice not to run you software as root, also in Docker.

Reply:
Although it does seem to add the user to group root. That is probably not a good idea.